src/main/java/org/reactome/addlinks/fileprocessors/FileProcessor.java

```diff
@@ -58,7 +55,7 @@ public FileProcessor(String processorName)
 	 * This assumes that the file contains a mapping, and in this context, it probably does.
 	 * @return
 	 */
-	public abstract Map<String, T> getIdMappingsFromFile();
+	public abstract Map<String, T> getIdMappingsFromFile() throws IOException;
```

Comment:
Does this break any compatibility with existing `FileProcessor` child classes that don't throw an `IOException`?

Reply:
I'm not sure, but I've run the AddLinks process without any issues. What kind of compatibility issues might you expect?

Reply:
I was just thinking since this is an abstract method, it's signature affects all its child classes.  So, it's important to ensure all its child classes will also throw an `IOException` when implementing this method.

Reply:
I don't think they do. Only BiomartFileProcessor can throw it since it draws all files from a directory and then reads them. The other ones draw from a specific file location that is configured outside of the class, so they don't throw the `IOException`

Reply:
Look at the usage for `GlobbedFileProcessor` (https://github.com/reactome/add-links/blob/83bd74bc3edb1cecad9870daed7121416cb7f261/src/main/java/org/reactome/addlinks/fileprocessors/GlobbedFileProcessor.java) to see a file processor that processes all files that match a file glob. Might make the processing of Biomart files easier...

Reply:
UniprotFileProcessor, KeggFileProcessor, EnsemblFileProcessor all extend the GlobbedFileProcessor.

Reply:
Actually, I looked into this more and throwing an exception on a method that gets overridden is okay.  It doesn't require the method in the child class to also throw an exception: https://stackoverflow.com/questions/17053063/java-abstract-classes-which-throw.

I am curious why the `throws IOException` was added on the abstract class' method, though?

Reply:
I believe that was an IDE correction after the IOException needed to be thrown from the class

Reply:
Ah okay, yeah - I just tried on my local copy of the repository and it says you can't override a method to throw an exception unless the method you're overriding also throws the same exception (or a child of the same exception).

Reply:
I inserted try/catches into the BiomartFileProcessor code, so now the IOException isn't thrown for FileProcessor